Fix flaky test addSnapshotLifecyclePolicy

andreidan opened the pull request:
Fixes #46021 

<!--
Thank you for your interest in and contributing to Elasticsearch! There
are a few simple things to check before submitting your pull request
that can help with the review process. You should delete these items
from your submission, but they are here to help bring them to your
attention.
-->

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

andreidan committed:
testAddSnapshotLifecyclePolicy create policy once

Drop async call to put the policy as we are not waiting for the result
as we are synchronously creating it already (the async nature of the
second create request made this test flaky as we're asserting the
version of the policy to be 1)

andreidan committed:
testAddSnapshotLifecyclePolicy drop unused variables

elasticmachine commented:
Pinging @elastic/es-core-features

dakrone commented on client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java:
```diff
@@ -797,32 +797,11 @@ public void testAddSnapshotLifecyclePolicy() throws Exception {
         assertTrue(putAcknowledged);
 
-        // tag::slm-put-snapshot-lifecycle-policy-execute-listener
-        ActionListener<AcknowledgedResponse> putListener =
```
These are here because they are used as snippets to generate the high level rest client documentation (that's what the surrounding `// tag::etcetc` and `// end::etcetc` are for, so I don't think we should remove these

dakrone commented on client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java:
```diff
@@ -858,18 +838,7 @@ public void onFailure(Exception e) {
         SnapshotLifecyclePolicyMetadata policyMeta =
             getResponse.getPolicies().get("policy_id"); // <1>
         long policyVersion = policyMeta.getVersion();
-        long policyModificationDate = policyMeta.getModifiedDate();
```
Even though these appear unused, they are documenting how to retrieve things from the policy metadata, so we should leave them. See `get_snapshot_lifecycle_policy.asciidoc` for the callout usages

dakrone commented on client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java:
```diff
@@ -797,32 +797,11 @@ public void testAddSnapshotLifecyclePolicy() throws Exception {
-        // end::slm-put-snapshot-lifecycle-policy-execute-listener
-
-        // tag::slm-put-snapshot-lifecycle-policy-execute-async
-        client.indexLifecycle().putSnapshotLifecyclePolicyAsync(request,
```
I think rather than remove this (which we need for documentation), we should relax or remove the assertion to not trip on the policy version.

dakrone reviewed (changes requested):
Thanks for taking a look at this Andrei, I left comments about what these unused blocks of code are for and why we shouldn't remove them.

andreidan committed:
Revert "testAddSnapshotLifecyclePolicy drop unused variables"

This reverts commit af424496cf603ec0f80fdb764ad8fea4dbcb7977.

andreidan committed:
Revert "testAddSnapshotLifecyclePolicy create policy once"

This reverts commit 32fa9d7cab17d2945af404bbd245542792a633cd.

andreidan committed:
addSnapshotLifecyclePolicy drop version assertion

This drops the assertion on the policy version (which was pinned to 1L)
as we want to execute both put policy apis (sync and async) for
documentation purposes. This will sometime (depending on the async
call) yield a version of 2L. Waiting for the async call to always
complete could be an option but the test is already rather slow and it's
a bit of an overkill as we're already verifyin the policy was created.

dakrone reviewed (approved):
LGTM

andreidan commented:
retest this please
